pkg/docker/config/config.go

```diff
@@ -275,6 +289,11 @@ func findAuthentication(registry, path string, legacyFormat bool) (string, strin
 		return getAuthFromCredHelper(ch, registry)
 	}
 
+	// Use the global credential helper, if set
+	if auths.CredsStore != "" {
+		return getAuthFromCredHelper(auths.CredsStore, registry)
+	}
```

Comment:
FIXME (I need to update/research): Is this right, or should we use existing entries in the config file first?

Reply:
> In the docker cli, this always overrides the file-based fetch and save whenever set (docker always decides on which store is appropriate for a registry and [returns only that store](https://github.com/docker/cli/blob/master/cli/command/registry/login.go#L143) - the helper-based store), so the same behaviour is implemented — it short circuits.